ql/src/java/org/apache/hadoop/hive/ql/exec/tez/HiveSplitGenerator.java

```diff
@@ -154,6 +173,107 @@ private void prepare(InputInitializerContext initializerContext) throws IOExcept
     LOG.info("SplitLocationProvider: " + splitLocationProvider);
   }
 
+  /**
+   * SplitSerializer is a helper class for taking care of serializing splits to the tez scratch dir
+   * when a size criteria defined by "hive.tez.input.fs.serialization.threshold" is met.
+   * It utilizes an ExecutorService for parallel writes to prevent a single split write operation
+   * becoming the bottleneck (as write() is called from a loop currently).
+   */
+  class SplitSerializer implements AutoCloseable {
```

Comment:
private static final

Reply:
this could not be either private or static or final
1. testing it from TestHiveSplitGenerator (private)
2. jobConf cannot be accessed from static context (static)
3. it's inherited in unit test (final)